tests/Feature/FlightControllerTest.php

```diff
@@ -2,547 +2,570 @@
+
+    $emptyDestinationData = [
+        'airline_id' => 1,
+        'origin_city_id' => 1,
+        'destination_city_id' => NULL,
+        'departure_time' => '2023-09-11T11:18',
+        'arrival_time' => '2023-09-11T15:19'
+    ];
+    $this->json('POST', action(StoreFlightController::class), $emptyDestinationData)
+        ->assertStatus(422)
+        ->assertJsonValidationErrors('destination_city_id');
+
+    $emptyDepartureData = [
+        'airline_id' => 1,
+        'origin_city_id' => 1,
+        'destination_city_id' => 2,
+        'departure_time' => NULL,
+        'arrival_time' => '2023-09-11T15:19'
+    ];
+    $this->json('POST', action(StoreFlightController::class), $emptyDepartureData)
+        ->assertStatus(422)
+        ->assertJsonValidationErrors('departure_time');
+
+    $emptyArrivalData = [
+        'airline_id' => 1,
+        'origin_city_id' => 1,
+        'destination_city_id' => 2,
+        'departure_time' => '2023-09-11T15:19',
+        'arrival_time' => NULL
+    ];
+    $this->json('POST', action(StoreFlightController::class), $emptyArrivalData)
+        ->assertStatus(422)
+        ->assertJsonValidationErrors('arrival_time');
+
+    $emptyData = [
+        'airline_id' => NULL,
+        'origin_city_id' => NULL,
+        'destination_city_id' => NULL,
+        'departure_time' => NULL,
+        'arrival_time' => NULL
+    ];
+    $this->json('POST', action(StoreFlightController::class), $emptyData)
+        ->assertStatus(422)
+        ->assertJsonValidationErrors('airline_id')
+        ->assertJsonValidationErrors('origin_city_id')
+        ->assertJsonValidationErrors('destination_city_id')
+        ->assertJsonValidationErrors('departure_time')
+        ->assertJsonValidationErrors('arrival_time');
+});
+
+it('creates a flights with valid parameters', function (){
+    City::factory(2)->create();
+    Airline::factory()->create();
+
+    $flightData = [
+        'airline_id' => 1,
+        'origin_city_id' => 1,
+        'destination_city_id' => 2,
+        'departure_time' => '2023-09-11T11:18',
+        'arrival_time' => '2023-09-11T15:19'
+    ];
+
+    $this->json('POST', action(StoreFlightController::class), $flightData)
+        ->assertStatus(200)
+        ->assertJson([
+            'status' => 200,
+            'success' => true,
+            'data' => [
+                'departure_time' => $flightData['departure_time'],
+                'arrival_time' => $flightData['arrival_time'],
+            ]
+        ]);
+});
+
+it('tries to create a flight with invalid city id but fails', function () {
+    City::factory()->create();
+    Airline::factory()->create();
+
+    $this->json('POST', action(StoreFlightController::class), [
+        'airline_id' => 1,
+        'origin_city_id' => 999,
+        'destination_city_id' => 1,
+        'departure_time' => '2023-09-11T11:18',
+        'arrival_time' => '2023-09-11T15:19'
+    ])
+        ->assertStatus(422)
```

Comment:
assertUnprocessable, 
PEST gives you a function with the name of every return code, so you can use them for readability
https://laravel.com/docs/10.x/http-tests#response-assertions